File: 1624-largest-substring-between-two-equal-characters/1624-largest-substring-between-two-equal-characters.py
class Solution(object):
    def maxLengthBetweenEqualCharacters(self, s):
        seen = [-1] * 26
        ans = -1

        for i in range(len(s)):
            if seen[ord(s[i]) - ord('a')] != -1:
                ans = max(ans, i - seen[ord(s[i]) - ord('a')] - 1)
                continue
            seen[ord(s[i]) - ord('a')] = i

        return ans
        

File: 1624-largest-substring-between-two-equal-characters/1624-largest-substring-between-two-equal-characters.java
class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        int[] freq = new int[26];
        Arrays.fill(freq, -1);
        int ans = -1;

        for(int i = 0; i<s.length(); i++) {
            Character c = s.charAt(i);
            if(freq[c - 'a'] != -1) {
                ans = Math.max(ans, i - freq[c - 'a'] - 1);
                continue;
            }
            freq[c - 'a'] = i;
        }

        return ans;
    }
}